### PR TITLE
#986 Implement catalog management using grids.

### PR DESCRIPTION
**Pull request by beghelli**

### CODE REVIEW

File: classes/monograph/PublishedMonographDAO.inc.php [hunk 1]
```diff
@@ -381,6 +278,132 @@ function getSortDirection($sortOption) {
...
+					break;
+				case ASSOC_TYPE_SERIES:
+					$series = DAORegistry::getDAO('SeriesDAO')->getById($assocId);
+					$sortOption = $series->getSortOption() ? $series->getSortOption() : $this->getDefaultSortOption();
+					break;
+				case ASSOC_TYPE_CATEGORY:
+					$category = DAORegistry::getDAO('CategoryDAO')->getById($assocId);
+					$sortOption = $category->getSortOption() ? $category->getSortOption() : $this->getDefaultSortOption();
+					break;
+			}
```
Comment (asmecher) [inline classes/monograph/PublishedMonographDAO.inc.php]: Suggest adding: `default: assert(false);`

File: classes/monograph/PublishedMonographDAO.inc.php [hunk 2]
```diff
@@ -381,6 +278,132 @@ function getSortDirection($sortOption) {
...
+			$sortDirection = $this->getSortDirection($sortOption);
+		}
+
+		// If no associated object is passed, we will get all monographs, featured/new released to the press or not.
+		$assocType = (int) $assocType?$assocType:ASSOC_TYPE_PRESS;
+		$assocId = (int) $assocId?$assocId:$pressId;
+		$params = array_merge(
+			array(REALLY_BIG_NUMBER),
+			$this->getFetchParameters(),
+			array(
```
Comment (asmecher) [inline classes/monograph/PublishedMonographDAO.inc.php]: Here and below, wherever possible, add `(int)` casts to database parameters.

File: classes/monograph/PublishedMonographDAO.inc.php [hunk 3]
```diff
@@ -381,6 +278,132 @@ function getSortDirection($sortOption) {
...
+					$category = DAORegistry::getDAO('CategoryDAO')->getById($assocId);
+					$sortOption = $category->getSortOption() ? $category->getSortOption() : $this->getDefaultSortOption();
+					break;
+			}
+			$sortBy = $this->getSortBy($sortOption);
+			$sortDirection = $this->getSortDirection($sortOption);
+		}
+
+		// If no associated object is passed, we will get all monographs, featured/new released to the press or not.
+		$assocType = (int) $assocType?$assocType:ASSOC_TYPE_PRESS;
```
Comment (asmecher) [inline classes/monograph/PublishedMonographDAO.inc.php]: Watch out -- the `(int)` cast gets applied to `$assocType`, not the result of the ternary statement.

File: controllers/grid/manageCatalog/CatalogMonographsGridHandler.inc.php [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+	function getRowInstance() {
+		import('controllers.grid.manageCatalog.CatalogMonographsGridRow');
+		return new CatalogMonographsGridRow();
+	}
+
+	/**
+	 * @copydoc GridHandler::getFilterSelectionData()
+	 */
+	function getFilterSelectionData($request) {
+		$filter = array();
```
Comment (asmecher) [inline controllers/grid/manageCatalog/CatalogMonographsGridHandler.inc.php]: I prefer...

    return array(
        'searchText' => $request->getUserVar('searchText'),
        'featured' => $request->getUserVar('featured'),
        'newReleased' => $request->getUserVar('newReleased'),
    );

...a little better. But just a preference.

File: classes/monograph/PublishedMonographDAO.inc.php (hunk 3 above)
Comment (beghelli) [inline classes/monograph/PublishedMonographDAO.inc.php]: oups, fixing.

**Comment by asmecher** [comment]
Also, I think you're already aware of this, but the Travis build failed.

**Comment by asmecher** [comment]
Committed directly -- thanks, @beghelli!

**Pull request closed without merging**